mlflow/shap.py

```diff
@@ -253,6 +260,8 @@ def predict_function(X) -> pred:
     _log_matplotlib_figure(fig, _SUMMARY_BAR_PLOT_FILE_NAME, artifact_path)
     plt.close(fig)
 
+    mlflow.utils.autologging_utils.AUTOLOGGING_INTEGRATIONS = original_config_dict
```

Comment:
Try to iterate over all integrations `mlflow.integration.autolog(disable=True)`.

Then restore enabled ones.